functests/cluster_upgrade_test.go

```diff
@@ -0,0 +1,76 @@
+package functests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/openshift/ocs-operator/functests"
+
+	deploymanager "github.com/openshift/ocs-operator/pkg/deploy-manager"
+)
+
+var _ = Describe("Cluster upgrade", func() {
+	flag.Parse()
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+	})
+
+	Describe("ocs", func() {
+
+		BeforeEach(func() {
```

Comment:
I think we need a corresponding `AfterEach()` that ensures we return to the expected ocs-operator version before allowing additional tests to execute. Otherwise, in the event that the upgrade test fails, we might continue testing with the wrong ocs-operator version installed.

Reply:
Added check to detect failure in the upgrade test and cleanup and reinstall the cluster for other tests to continue on.